src/Core/ExecutionPathTracer/ExecutionPath.cs

```diff
@@ -0,0 +1,131 @@
+        /// </summary>
+        public Operation[] Operations { get; set; }
+
+        /// <summary>
+        /// Serializes <c>ExecutionPath</c> into its JSON representation.
+        /// </summary>
+        /// <param name="prettyPrint">
+        /// Pretty prints the JSON (i.e. with white space and indents) if true (default = false).
+        /// </param>
+        public string ToJson(bool prettyPrint = false)
+        {
+            var options = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                IgnoreNullValues = true,
+                WriteIndented = prettyPrint,
+            };
+            return JsonSerializer.Serialize(this, options);
+        }
+    }
+
+    /// <summary>
+    /// Represents a qubit resource used in an execution path.
+    /// </summary>
+    public class QubitDeclaration
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QubitDeclaration"/> class.
+        /// </summary>
+        /// <param name="id">
+        /// Id of qubit.
+        /// </param>
+        /// <param name="numChildren">
+        /// Number of associated classical registers.
+        /// </param>
+        public QubitDeclaration(int id, int? numChildren = null)
+        {
+            this.Id = id;
+            this.NumChildren = numChildren;
+        }
+
+        /// <summary>
+        /// Id of qubit.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Number of associated classical registers.
+        /// </summary>
+        public int? NumChildren { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an operation used in an execution path.
+    /// </summary>
+    public class Operation
+    {
+        /// <summary>
+        /// Label of gate.
+        /// </summary>
+        public string Gate { get; set; }
+
+        /// <summary>
+        /// Non-qubit arguments provided to gate.
+        /// </summary>
+        public string ArgStr { get; set; }
+
+        /// <summary>
+        /// Group of operations for each classical branch.
+        /// </summary>
+        /// <remarks>
+        /// Currently not used as this is intended for classically-controlled operations.
+        /// </remarks>
+        public List<List<Operation>> Children { get; set; }
+
+        /// <summary>
+        /// True if operation is a controlled operations.
+        /// </summary>
+        public bool Controlled { get; set; }
+
+        /// <summary>
+        /// True if operation is an adjoint operations.
+        /// </summary>
+        public bool Adjoint { get; set; }
+
+        /// <summary>
+        /// List of control registers.
+        /// </summary>
+        public List<Register> Controls { get; set; } = new List<Register>();
+
+        /// <summary>
+        /// List of target registers.
+        /// </summary>
+        public List<Register> Targets { get; set; } = new List<Register>();
```

Comment:
Same here, consider using `IEnumerable` instead of `List` for the property type. Of course, using `new List` is still totally fine. 😊

Reply:
Thanks for the tip! Really appreciate it :D